packages/puppeteer-extra-plugin-stealth/evasions/navigator.hardwareConcurrency/index.js

```diff
@@ -0,0 +1,29 @@
+'use strict'
+
+const {PuppeteerExtraPlugin} = require('puppeteer-extra-plugin')
+
+/**
+ * Allows faking CPU core number. This allows creating different profiles.
+ * Will overwrite `navigator.hardwareConcurrency` property.
+ */
+class Plugin extends PuppeteerExtraPlugin {
+    constructor(opts = {}) {
+        super(opts)
+    }
+
+    get name() {
+        return 'stealth/evasions/navigator.hardwareConcurrency'
+    }
+
+
+    async onPageCreated(page) {
+        await page.evaluateOnNewDocument((cores) => {
+            let originalCores = navigator.hardwareConcurrency;
```

Comment:
- `originalCores` can be `const` as the value is never changing
- I'd feel more comfortable if we do a quick check if `cores` is defined and if not just `return` without any modification (also no need for `originalCores` then)

Reply:
Would `this.opts.cores || navigator.hardwareConcurrency` be okay ? or something like
getattr(this.opts, 'cores', navigator.hardwareConcurrency)

Reply:
What I meant is let's not overwrite the property at all if theres no need. :-)

Example:

```js
async onPageCreated(page) {
  if (!this.opts.cores || !Number.isInteger(this.opts.cores)) {
    return
  }

  await page.evaluateOnNewDocument((cores) => {
    Object.defineProperty(Object.getPrototypeOf(navigator), 'hardwareConcurrency', { get: () => cores });
  }, this.opts.cores)
}
```

Reply:
Ah. That makes a lot more sense. Made the change.